app/src/main/java/sdp/moneyrun/user/User.java

```diff
@@ -30,228 +38,280 @@ public void setMaxScoreInGame(int maxScoreInGame) {
         }
 
-        /**
-         *
-         * @return the name of the user
-         */
-        public String getName() {
-            return name;
-        }
+        return this.friendIdList.remove(friendId);
+    }
 
-        /**
-         *
-         * @return the number of games the user played
-         */
-        public int getNumberOfPlayedGames() {
-            return numberOfPlayedGames;
-        }
+    /**
+     * @return number of games in which the user died
+     */
+    public int getNumberOfDiedGames() {
+        return numberOfDiedGames;
+    }
 
+    /**
+     * @return the unique user id
+     */
+    public String getUserId() {
+        return userId;
+    }
 
-        @Override
-        public boolean equals(Object o) {
-            if (this == o) return true;
-            if (o == null || getClass() != o.getClass()) return false;
-            User user = (User) o;
-            return userId.equals(user.userId) &&
-                    numberOfPlayedGames == user.numberOfPlayedGames &&
-                    numberOfDiedGames == user.numberOfDiedGames &&
-                    name.equals(user.name) &&
-                    address.equals(user.address);
-        }
+    /**
+     * @return the adress of the user
+     */
+    public String getAddress() {
+        return address;
+    }
 
-        @Override
-        public int hashCode() {
-            return Objects.hash(userId, name, address, numberOfPlayedGames, numberOfDiedGames);
-        }
+    /**
+     * @return the name of the user
+     */
+    public String getName() {
+        return name;
+    }
 
-        /**
-         * Updates user values in the firebase database
-         * @param dbChange
-         */
-        private void dbUpdate(boolean dbChange){
-            if(dbChange){
-                UserDatabaseProxy pdb = new UserDatabaseProxy();
-                pdb.putUser(this);
-            }
-        }
+    /**
+     * @return the number of games the user played
+     */
+    public int getNumberOfPlayedGames() {
+        return numberOfPlayedGames;
+    }
+
+    public List<String> getFriendIdList(){
+        return new ArrayList<>(this.friendIdList);
+    }
 
 
-        /**
-         *
-         * @param question
-         * @return the answer of the question asked
-         */
-        public String ask(String question){
-            String answer = "";
-            //TODO: display question on  user's screen and store the response
-            return answer;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        User user = (User) o;
+        return userId.equals(user.userId) &&
+                numberOfPlayedGames == user.numberOfPlayedGames &&
+                numberOfDiedGames == user.numberOfDiedGames &&
+                name.equals(user.name) &&
+                address.equals(user.address);
+    }
```

Comment:
Do you have to add the friends list to the .equals() criteria ? I'm not sure, but I wanted to check with you

Reply:
Yes, this is important, thank you

Reply:
done